src/sinks/databend/config.rs

```diff
@@ -0,0 +1,209 @@
+pub struct DatabendConfig {
+    /// The endpoint of the Databend server.
+    #[configurable(metadata(docs::examples = "http://localhost:8000"))]
+    pub endpoint: UriSerde,
+
+    /// The table that data will be inserted into.
+    #[configurable(metadata(docs::examples = "mytable"))]
+    pub table: String,
+
+    /// The database that contains the table that data will be inserted into.
+    #[configurable(metadata(docs::examples = "mydatabase"))]
+    #[serde(default = "DatabendConfig::default_database")]
+    pub database: String,
+
+    #[configurable(derived)]
+    #[serde(default = "Compression::gzip_default")]
+    pub compression: Compression,
+
+    #[serde(flatten)]
+    pub encoding: EncodingConfigWithFraming,
+
+    #[configurable(derived)]
+    #[serde(default)]
+    pub batch: BatchConfig<RealtimeSizeBasedDefaultBatchSettings>,
+
+    #[configurable(derived)]
+    pub auth: Option<Auth>,
+
+    #[configurable(derived)]
+    #[serde(default)]
+    pub request: TowerRequestConfig,
+
+    #[configurable(derived)]
+    pub tls: Option<TlsConfig>,
+
+    #[configurable(derived)]
+    #[serde(
+        default,
+        deserialize_with = "crate::serde::bool_or_struct",
+        skip_serializing_if = "crate::serde::skip_serializing_if_default"
+    )]
+    pub acknowledgements: AcknowledgementsConfig,
+}
+
+impl GenerateConfig for DatabendConfig {
+    fn generate_config() -> toml::Value {
+        toml::from_str(
+            r#"endpoint = "http://localhost:8000"
+            encoding.codec = "json"
+            table = "default"
+            database = "default"
+        "#,
+        )
+        .unwrap()
+    }
+}
+
+impl DatabendConfig {
+    pub(super) fn build_client(&self, cx: &SinkContext) -> crate::Result<HttpClient> {
+        let tls = TlsSettings::from_options(&self.tls)?;
+        let client = HttpClient::new(tls, cx.proxy())?;
+        Ok(client)
+    }
+
+    fn default_database() -> String {
+        "default".to_string()
+    }
+}
+
+#[async_trait::async_trait]
+impl SinkConfig for DatabendConfig {
+    async fn build(&self, cx: SinkContext) -> crate::Result<(VectorSink, Healthcheck)> {
+        let auth = self.auth.choose_one(&self.endpoint.auth)?;
+        let endpoint = self.endpoint.with_default_parts();
+        let config = DatabendConfig {
+            auth: auth.clone(),
+            ..self.clone()
+        };
+        let health_client =
+            DatabendAPIClient::new(self.build_client(&cx)?, endpoint.clone(), auth.clone());
+        let healthcheck = select_one(health_client).boxed();
+
+        let request_settings = self.request.unwrap_with(&TowerRequestConfig::default());
+        let batch_settings = self.batch.into_batcher_settings()?;
+
+        let database = config.database;
+        let table = config.table;
+        let client = DatabendAPIClient::new(self.build_client(&cx)?, endpoint, auth);
+
+        let (framer, serializer) = self.encoding.build(SinkType::StreamBased)?;
```

Comment:
```suggestion
        let encoding_config = EncodingConfigWithFraming::new(
            Some(FramingConfig::NewlineDelimited),
            SerializerConfig::Json(JsonSerializerConfig),
            self.encoding,
        );
        let (framer, serializer) = encoding_config.build(SinkType::StreamBased)?;
```

Continuing off the above suggestion, you should be able to do something like this to build the encoding configuration from the user provided `Transformer` and internally selecting Json and newline delimiters (`None` for the framer should [do this regardless](https://github.com/vectordotdev/vector/blob/master/src/codecs/encoding/config.rs#L103-L104), for `StreamBased` but why not by specific).